src/modules/orders/index.ts

```diff
@@ -67,13 +143,479 @@ class OrdersService {
+  Match engine is design for meet user need, and we have order amount in cell data to define this need.
+  So for ask order, you need to buy enough CKB amount.We don't care it's SUDT amount left.
+  For bid order, you need to buy enough SUDT amount. We don't care it's ckb amount left.
+  For match there have two condition: 1. full matched 2. partially matched.
+  Full matched means two orders need both can be meet.
+  In other words, ask order's need ckb amount is equal to bid order can pay for it's SUDT value.
+  Partially matched means only one direction's need can be meet.So one order will be deal done,
+  another will have left.
+  */
+  private startMatchAndReturnOutputs(bidOrderList: Array<OrderDto>, askOrderList: Array<OrderDto>): any {
+    let askMatchOrder = askOrderList[0]
+    let bidMatchOrder = bidOrderList[0]
+    const { price: askPrice } = askMatchOrder
+    const { price: bidPrice } = bidMatchOrder
+
+    const askCapacityPrice = (askPrice * this.shannonsRatio) / this.priceRatio
+    const bidCapacityPrice = (bidPrice * this.shannonsRatio) / this.priceRatio
+
+    const askOrderOutput = JSON.parse(askMatchOrder.output)
+    const bidOrderOutput = JSON.parse(bidMatchOrder.output)
+    const bidOriginalScript = {
+      lock: bidOrderOutput.lock,
+      type: bidOrderOutput.type,
+    }
+    const askOriginalScript = {
+      lock: askOrderOutput.lock,
+      type: askOrderOutput.type,
+    }
+
+    //When ask price is greater than bid price, they can't match.
+    if (askPrice > bidPrice) {
+      //If before we have matched some order and current order is partially matched, we need to add it to outputs.
+      if (this.outputsCells.length > 0) {
+        if (bidMatchOrder.part) {
+          this.pushInputCells(bidMatchOrder.id, undefined)
+          this.pushOutputsCellAndData(
+            { capacity: bidOrderOutput.capacity, data: bidOrderOutput.data },
+            bidOriginalScript,
+          )
+        }
+
+        if (askMatchOrder.part) {
+          this.pushInputCells(askMatchOrder.id, undefined)
+          this.pushOutputsCellAndData(
+            { capacity: askOrderOutput.capacity, data: askOrderOutput.data },
+            askOriginalScript,
+          )
+        }
+
+        return this.outputsCells
+      } else {
+        console.info('No match')
+        return []
+      }
+    } else {
+      // We calculate deal price by a simmplest strategy, use average price.After all calculation we use deal,so forget order price.
+      const dealPrice: bigint = (askCapacityPrice + bidCapacityPrice) / BigInt('2')
+
+      const bidSudtAmount: bigint = parseOrderData(bidOrderOutput.data).sudtAmount
+      const bidSudtOrderAmount: bigint = parseOrderData(bidOrderOutput.data).orderAmount
+      const bidActualSpendCapacityAmount: bigint = (dealPrice * bidSudtOrderAmount) / this.shannonsRatio
+      const bidOriginalCapacityAmount: bigint = BigInt(bidOrderOutput.capacity)
```

Comment:
They can be organized in a object named `bid` so the reference can be more clear, as `bid.SudtAmount`, bid.SudtOrderAmount`